src/main/java/org/jboss/logmanager/handlers/AsyncHandler.java

```diff
@@ -146,6 +146,10 @@ protected void doPublish(final ExtLogRecord record) {
             // Copy the MDC over
             record.copyMdc();
         }
+        if (record.getCallerThread() != null && record.getCallerThread().equals(thread)) {
```

Comment:
This can be simplified to `if (record.getCallerThread() == thread) {` since threads should always be compared by identity using `==`, and using `==` also covers `null`.